### ISSUE TITLE
[4.0.0-beta1] BIOS or game files with upper case extension aren't found at launch of a game anymore (fba_libretro, mame, ?)

### ISSUE THREAD

**Issue by Voljega** [recalbox/recalbox-os#722]
Game launch, display switch to 1080p (I didn't modify my display conf at all after install), and then it goes back to EmulationStation without showing any picture
tried with Metal Slug 1,6 and Last Blade
Off course I Have the neogeo rom into the roms folder

Message for the unconfigured xbox pad does'nt appear either

Other arcade games are still launched with fba_libretro (and  unconfigured xbox pad message is showing)

**Comment by substring** [comment]
I have just tried last blade, it worked fine for me

**Comment by tlanks** [comment]
U need neogeo.zip

**Comment by Voljega** [comment]
@tlanks yeah thank you very much that's what I said in the first message that I have it...

OK well I noticed something which may explain the problem and I'll have to try :
- I copied my entire roms directory from 3.3.0b16 to 4.0.0 after installation of the new version
- I have now two directories fba_libretro and fbalibretro, the name changed between the two versions

When I use the new version, fba roms appears in the menu although being in the ancient directory, but gamelist and related images aren't found apparently.

So I'm guessing maybe there is something fishy, such as ancient directory name beeing still used in the systems description but the new one being used everywhere else, beacause the roms shouldn't even show in recalbox if they are in the old named folder.

Maybe the neogeo rom is searched into the new directory and not found or something, I'll try renaming the old directory

**Comment by tlanks** [comment]
`<system>
        <fullname>FBA LIBRETRO</fullname>
        <name>fba_libretro</name>
        <path>/recalbox/share/roms/fba_libretro</path>
        <extension>.zip .ZIP .fba .FBA</extension>
        <command>python /usr/lib/python2.7/site-packages/configgen/emulatorlauncher.pyc %CONTROLLERSCONFIG% -system %SYSTEM% -rom %ROM% -emulator %EMULATOR% -core %CORE%</command>
        <platform>arcade</platform>
        <theme>fba_libretro</theme>
	<emulators>
	    <emulator name="libretro">
	        <cores>
		    <core>fba</core>
		</cores>
	    </emulator>
	</emulators>
    </system>`

**Comment by tlanks** [comment]
<system>
        <fullname>FBA LIBRETRO</fullname>
        <name>fba_libretro</name>
        <path>/recalbox/share/roms/fba_libretro</path>
        <extension>.zip .ZIP .fba .FBA</extension>
        <command>python /usr/lib/python2.7/site-packages/configgen/emulatorlauncher.pyc %CONTROLLERSCONFIG% -system %SYSTEM% -rom %ROM% -emulator %EMULATOR% -core %CORE%</command>
        <platform>arcade</platform>
        <theme>fba_libretro</theme>
	<emulators>
	    <emulator name="libretro">
	        <cores>
		    <core>fba</core>
		</cores>
	    </emulator>
	</emulators>
    </system>

**Comment by tlanks** [comment]
[es_systems.cfg](https://github.com/recalbox/recalbox-buildroot/commit/6d5ffa497ba990a28cb28a7b38b7b3e3df3c08b0)

**Comment by Voljega** [comment]
Your link doesn't seem to point toward the right file (which I could'nt find in the new version where it used to be)

**Comment by substring** [comment]
@Voljega I suspect you just copied your rom folder + scraping to your v4.0.0. The roms folder for fba_libretro didn't change. But, there were a few internal changes concerning the scraping folder. All has been unified to fba_libretro.

Check https://github.com/recalbox/recalbox-configgen/pull/9 and https://github.com/recalbox/recalbox-buildroot/pull/74

**Comment by Voljega** [comment]
I just copied my whole old roms folder over my whole new roms folder :
I have now two different folders fba_libretro and fbalibretro in my roms folder... 
One with all the roms (so this is my old one from 3.3.0beta16) and one with gamelist.xml (so this is the new one from 4.4.0beta1 because gamelists weren't found there in 3.3.0beta16)

And still ES shows my roms (it shouldn't in my opinion) but not my scrape...

I'll try tonight to merge the two folders to the right new one and see what happens regarding ES and neogeogames

**Comment by paradadf** [comment]
It might be related to issue https://github.com/recalbox/recalbox-os/issues/536

**Comment by substring** [comment]
Check my links @paradadf, it's been solved a while ago

**Comment by paradadf** [comment]
@substring I was not saying, that it was a problem of v4.0.0 beta 1, but a problem with @Voljega´s migration from v3.

**Comment by Voljega** [comment]
Yeah I'm not realy worried for the scrape issue, it's likely a simple folder name issue.
But the fact that my neogeo games aren't working anymore, I don't really understand it...

**Comment by Voljega** [comment]
Yeah scrape wasn't an issue after all, just a folder name problem.

But PGM aren't launched either on 4.4.0b1
I checked on my 3.3.0b16 all those games are working perfectly.

So I was able to find the source of the problem / regression : my BIOS/drivers files in fba_libretro are renamed to .ZIP (instead of .zip) to hide them from ES menu with the proper modifications in es_systems.cfg.

In 4.4.0b1 I didn't modified my es_systems.cfg for the moment so the BIOS/drivers files appears in ES menu all right.

But at launch of a game using an additional BIOS, recalbox seems to not be able anymore to find the BIOS file named with an upper case extension like .ZIP, renaming them to .zip correct the issue and the games are launched.

And the same problem seems to appear on MAME too !

So it's likely it appears on every other emulator

**Comment by digitalLumberjack** [comment]
There is no possibility to change that, as the internal ext4 partition is case sensitive (and will be, we won't change that as it would bring more bugs thant it would correct)
So the solutions are : change the .cue files to point to the good files, or use an external storage in FAT32.

**Comment by Voljega** [comment]
Well the test case is about arcade roms so this is not related to cue files at all.
And I'm using an external storage and it is indeed in FAT32.
So problem still there and closing the bug is a little early.

It worked perfectly fine in 3.3.0, so some kind of regression must have obviously been introduced.

**Comment by digitalLumberjack** [comment]
Hum sorry i copy past the response from the other issue.
But it's still the same if the issue is the uppercase in the rom name, it's not really an issue. Rename it in .zip and we're good.

**Comment by Voljega** [comment]
Using uppercase extensions is commonly used by Recalbox users to hide unwanted files, it's a shame to discard that functionnality so easily.

And iIt's obviously not the same issue given that the bug appears with roms stored on a FAT32 external storage

**Comment by digitalLumberjack** [comment]
I added a hide / show metadata to replace this trick, isn't it enough ?

**Comment by Voljega** [comment]
It's good but for arcade games you have to manually hide every bios files in every folder or sub folder you have, in my case this is something like 200-300 files...

**Comment by digitalLumberjack** [comment]
Didn't you manually renammed your 200 / 300 files to get the .ZIP in uppercase ?

**Comment by Voljega** [comment]
No I manually edited the extension of some 30 BIOS files and then copied them to every subfolder.

**Comment by digitalLumberjack** [comment]
Maybe you could manually add all the hidden bios in your gamelist.xml and copy the section for each subfolders.
What are those 30 bios ?

**Comment by Voljega** [comment]
The 30-some files for every BIOS/driver in mame or fba libretro full romset.

Off course I can do that and i will do it if needed but I'm not the average user either, some people won't be able to do it

**Comment by digitalLumberjack** [comment]
people only have one folder with their bios most of time

**Comment by digitalLumberjack** [comment]
could you try to remount your fat32 partition like : 
mount -t vfat -o remount,check=r /dev/yourexternalstorage /recalbox/share

**Comment by Voljega** [comment]
what should 'yourexternalstorage' be ?

because my external storage is named RxRoms but there's no /dev/RxRoms or /dev/usb0 /dev/usbxx when i look into the dir

**Comment by Voljega** [comment]
plouf plouf ?

**Comment by digitalLumberjack** [comment]
Use `cat /proc/partitions` and see which is your external storage (check the size, and it should be a /dev/sd*)

**Comment by Voljega** [comment]
cat /proc/partitions give me   
8        0  488386583 sda
8        1  488384000 sda1

My hardrive is 500 Go so hard to say but in /dev :
# ls -l sd*
brw-rw----    1 root     disk        8,   0 Jan  1  1970 sda
brw-rw----    1 root     disk        8,   1 Mar 29 11:07 sda1

I did a fresh install on Pi3 today so I asume the right on is sda1

Then mount -t vfat -o remount,check=r /dev/sda1 /recalbox/share didn't correct the uppercase problem

**Comment by substring** [comment]
@recalbox/recalbox-testers need you to try something :
- prepare a FAT32 USB key with some cue+bin files + any other rom
- comment line 37 in `/recalbox/scripts/recalbox-mount.sh` : https://github.com/recalbox/recalbox-buildroot/blob/0c76ad4bf849422301203a48b2b13dbb99297698/board/recalbox/fsoverlay/recalbox/scripts/recalbox-mount.sh#L37
- reboot
- mix cases for the .bin filename and its reference in the .cue name (no need to restart ES or recalbox)
- confirm this solves (or not) the case problem
- test for non regression for other roms

thanks guyz :+1:

**Comment by benutzer193** [comment]
Which ROMs are using cue bin? Never used it before.

**Comment by paradadf** [comment]
@substring I can confirm you that commenting line 37 doesn't solve the problem. It even gets worst, as I wasn't able to connect to my wifi network anymore until I uncommented that line.

**Comment by substring** [comment]
@benutzer193 PSX isos, PC Engine CD isos
@paradadf you're the first one telling me that. Been trying it myself, Nosfay did too, we had the expected result for isos. I'm surprises this impacts your wifi

**Comment by paradadf** [comment]
@substring Well then, I don't use external storages myself. If I'm the only one having that "issue", either I might have done something wrong or my settings were somehow inconvenient. It would be useful if others try it and confirm its working.

**Comment by substring** [comment]
@paradadf if you drop all on your SD, then of course you can't test what i suggested ;) you do need a FAT32 USB key for the steps i mentionned. If you left everything on your SD the share partition is in ext4, so we can't solve it unless heavily patching libretro or the cores themselves. And we're not really eager to do that ;)

**Comment by paradadf** [comment]
@substring Maybe I was 't clear enough. I don't use any external storage option BUT I used a FAT32 formatted USB to test this specific problem and try to help ;)

**Comment by digitalLumberjack** [comment]
Moving that to the backlog
